core/src/desul/atomics/SYCL.hpp

```diff
@@ -17,17 +17,69 @@ SPDX-License-Identifier: (BSD-3-Clause)
 
 namespace desul {
 
+// FIXME_SYCL We need to either use generic_space or figure out how to check for the
+// correct adress space in a SYCL-portable way.
+#if defined(__SYCL_DEVICE_ONLY__) && !defined(__NVPTX__)
```

Comment:
```suggestion
#ifdef __NVPTX__
```

Reply:
Why? I need both checks for `__SYCL_GenericCastToPtrExplicit_ToLocal` being available.

Reply:
https://github.com/kokkos/kokkos/blob/48beb0b7192d070030a118d6c375efc69cbe0b12/core/src/desul/atomics/Macros.hpp#L35-L37

Reply:
OK. I would consider restricting the SYCL atomics to device only more like a temporary solution but I don't feel overly strong about it. Do you want me to change?

Reply:
The code is guarded by https://github.com/kokkos/kokkos/blob/2307ce3271c92d9d21307f7159eea4c6858e5c59/core/src/desul/atomics/Compare_Exchange_SYCL.hpp#L19 so this is redundant

Reply:
I already removed it in [`2307ce3` (#4585)](https://github.com/kokkos/kokkos/pull/4585/commits/2307ce3271c92d9d21307f7159eea4c6858e5c59).